libs/core/src/main/java/org/opensearch/core/common/io/stream/BufferedChecksumStreamOutput.java

```diff
@@ -90,4 +100,80 @@ public void reset() throws IOException {
     public void resetDigest() {
         digest.reset();
     }
+
+    @Override
+    public void writeMap(@Nullable Map<String, Object> map) throws IOException {
+        Map<String, Object> newMap = new TreeMap<>(map);
+        writeGenericValue(newMap);
+    }
+
+    public <K, V> void writeMapOrdered(
+        Map<K, V> map,
+        final Writeable.Writer<K> keyWriter,
+        final Writeable.Writer<V> valueWriter,
+        Comparator<Map.Entry<K, V>> entryComparator
+    ) throws IOException {
+        writeVInt(map.size());
+        map.entrySet().stream().sorted(entryComparator).forEachOrdered(entry -> {
+            try {
+                keyWriter.write(this, entry.getKey());
+                valueWriter.write(this, entry.getValue());
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to write map values.", e);
+            }
+        });
+    }
```

Comment:
```
@Override
    public <K, V> void writeMap(final Map<K, V> map, final Writeable.Writer<K> keyWriter, final Writeable.Writer<V> valueWriter)
        throws IOException {
        writeVInt(map.size());
        Stream<Map.Entry<K, V>> sorted = map.entrySet().stream().sorted(Comparator.comparing(Objects::hashCode));
        sorted.forEachOrdered(entry -> {
            try {
                keyWriter.write(this, entry.getKey());
                valueWriter.write(this, entry.getValue());
            } catch (IOException e) {
                throw new RuntimeException("Failed to write map.", e);
            }
        });
    }
```
Wouldn't this work to sort the `Entry` based on the hashcode?

Reply:
We dont want to sort by hashcode since that doesnt give consistent ordering. The comparator being passed here would take care of ensuring ordering.